cdoc/cdoc_jar/src/main/java/org/cdoc4j/cli/Tool.java

```diff
@@ -0,0 +1,164 @@
+            String idCode = argParts[2];
+            
+            String fileName = nameFrom+".txt";
+
+            try {
+                File plain = new File(fileName);
+                if (!plain.createNewFile()) {
+                    System.out.println("File creation failed");
+                }
+                FileWriter writer = new FileWriter(fileName);
+                writer.write(nameTo);
+                writer.close();
+
+                encrypt(fileName, idCode);
+                
+                plain.delete();
+                Files.move(Paths.get(fileName+".cdoc"), Paths.get("cdoc_files/"+nameFrom+".cdoc"));
+                
+
+                response.status(200);
+                return "Encryption successful";
+
+            } catch (Exception e) {
+                response.status(500);
+                return "Exception occurred: " + e.getMessage();
+            }
+        });
+
+        // Health check endpoint
+        get("/health", (req, res) -> "API is up and running!");
+    }
+
+    public static void encrypt(String fileName, String idCode) throws Exception {
+
+        try {
+            // The total list of certificates to encrypt against.
+            HashSet<X509Certificate> recipients = new HashSet<>();
+
+            List<Path> files = new ArrayList<>();
+            HashSet<String> codes = new HashSet<>();
+
+            Path f = Paths.get(fileName);
+            if (Files.isRegularFile(f)) {
+                files.add(f);
+            } else {
+                fail(fileName + " is not a valid file");
+            }
+
+            if (IDCode.check(idCode)) {
+                codes.add(idCode);
+            } else {
+                fail(idCode + " is not a valid id code");
+            }
+
+            // Resolve certificates from LDAP
+            for (String code : codes) {
+                Collection<X509Certificate> c = LDAP.fetch(code);
+                if (!c.isEmpty()) {
+                    // Always filter LDAP certs
+                    Collection<X509Certificate> uc = filter_crypto_certs(c);
+                    recipients.addAll(uc);
+                }
+            }
+
+
+            if (recipients.isEmpty()) {
+                fail("must specify recipients for encryption");
+            }
+
+            // Check if -o is present
+            final File outfile;
+            // Single file
+            String fn = files.get(0).getFileName() + ".cdoc";
+            outfile = new File(fn);
+
+
+            if (outfile.exists())
+                fail("Output file " + outfile + " already exists");
+
+            CDOCBuilder cdoc = CDOC.builder();
+
+            // Recipients
+            for (X509Certificate r : recipients) {
+                cdoc.addRecipient(r);
+            }
+
+            // Source files
+            for (Path p : files) {
+                cdoc.addPath(p);
+            }
+
+            // Shoot
+            cdoc.buildToStream(Files.newOutputStream(outfile.toPath()));
```

Comment:
You have a buildToStream function, why not build the stream directly to response? This way you can also eliminate the requirement of having the 2 web services share file storage.